tests/js/spec/views/dashboardsV2/widgetBuilder/widgetBuilder.spec.tsx

```diff
@@ -190,4 +245,1168 @@ describe('WidgetBuilder', function () {
+            period: '24h',
+            utc: false,
+            project: [],
+            environment: [],
+          },
+        })
+      );
+    });
+  });
+
+  it('can update the title', async function () {
+    renderTestComponent({
+      query: {source: DashboardWidgetSource.DISCOVERV2},
+    });
+
+    const customWidgetLabels = await screen.findAllByText('Custom Widget');
+    // EditableText and chart title
+    expect(customWidgetLabels).toHaveLength(2);
+
+    userEvent.click(customWidgetLabels[0]);
+    userEvent.clear(screen.getByRole('textbox', {name: 'Widget title'}));
+    userEvent.type(
+      screen.getByRole('textbox', {name: 'Widget title'}),
+      'Unique Users{enter}'
+    );
+
+    expect(screen.queryByText('Custom Widget')).not.toBeInTheDocument();
+
+    expect(screen.getAllByText('Unique Users')).toHaveLength(2);
+  });
+
+  it('can add query conditions', async function () {
+    const {router} = renderTestComponent({
+      query: {source: DashboardWidgetSource.DISCOVERV2},
+    });
+
+    userEvent.type(
+      await screen.findByRole('textbox', {name: 'Search events'}),
+      'color:blue{enter}'
+    );
+
+    userEvent.click(screen.getByText('Select a dashboard'));
+    userEvent.click(screen.getByText('Test Dashboard'));
+    userEvent.click(screen.getByText('Add Widget'));
+
+    await waitFor(() => {
+      expect(router.push).toHaveBeenCalledWith(
+        expect.objectContaining({
+          pathname: '/organizations/org-slug/dashboard/2/',
+          query: {
+            displayType: 'table',
+            interval: '5m',
+            title: 'Custom Widget',
+            queryNames: [''],
+            queryConditions: ['color:blue'],
+            queryFields: ['count()'],
+            queryOrderby: '',
+            start: null,
+            end: null,
+            period: '24h',
+            utc: false,
+            project: [],
+            environment: [],
+          },
+        })
+      );
+    });
+  });
+
+  it('can choose a field', async function () {
+    const {router} = renderTestComponent({
+      query: {source: DashboardWidgetSource.DISCOVERV2},
+    });
+
+    expect(await screen.findAllByText('Custom Widget')).toHaveLength(2);
+
+    // No delete button as there is only one query.
+    expect(screen.queryByRole('button', {name: 'Remove query'})).not.toBeInTheDocument();
+
+    const countFields = screen.getAllByText('count()');
+    expect(countFields).toHaveLength(2);
+
+    userEvent.click(countFields[1]);
+    userEvent.type(countFields[1], 'last');
+    userEvent.click(screen.getByText('last_seen()'));
+
+    userEvent.click(screen.getByText('Select a dashboard'));
+    userEvent.click(screen.getByText('Test Dashboard'));
+    userEvent.click(screen.getByText('Add Widget'));
+
+    await waitFor(() => {
+      expect(router.push).toHaveBeenCalledWith(
+        expect.objectContaining({
+          pathname: '/organizations/org-slug/dashboard/2/',
+          query: {
```

Comment:
These tests that check that the builder redirects us with these query params are testing an "Add to Dashboard" flow, right? Should we nest them under a `describe` block and call that out?

Reply:
yes. I just copied the tests from the  old file and did not update the titles, but will do it 😉 thanks